cli/simulate_pixels.py

```diff
@@ -973,7 +972,7 @@ def save_results(event_times, is_first_batch, results, i_trig, i_mod=-1, light_o
                 if light.LIGHT_SIMULATED:
                     RangePush("sum_light_signals")
                     light_inc = light_sim_dat[batch_mask][itrk:itrk+sim.BATCH_SIZE]
-                    selected_track_id = track_ids[batch_mask][itrk:itrk+sim.BATCH_SIZE]
+                    selected_track_id = np.ascontiguousarray(selected_tracks["segment_id"])
```

Comment:
perhaps it makes more sense to cast it into a cupy array here `selected_track_id = cp.asarray(selected_tracks["segment_id"])`

Reply:
@marjoleinvannuland I'm really sorry... Actually it's better to change "track_ids[batch_mask][itrk:itrk+sim.BATCH_SIZE]" now "cp.array(selected_tracks["segment_id"])" to "segment_ids_arr[batch_mask][itrk:itrk+sim.BATCH_SIZE]" to make it consistent between charge and light.

Reply:
@marjoleinvannuland Meanwhile, I think we can remove this line (https://github.com/DUNE/larnd-sim/blob/e9fdcc0cfae33335648fba17abc2da8f310c63b0/cli/simulate_pixels.py#L764) in the same PR

Reply:
Ok, I updated the PR with the requested changes. The validation plots still look good.
![unique_evid_per_seg_count](https://github.com/DUNE/larnd-sim/assets/24715617/dff78cbb-962f-4e13-85bd-3a1a8fa185db)
![unique_evid_per_seg_heatmap](https://github.com/DUNE/larnd-sim/assets/24715617/f66b0899-2f34-4e2c-b210-01cb404eed7a)
[larndsim validation plots](https://github.com/DUNE/larnd-sim/files/15235376/final3_validations.pdf)